xcp_d/utils/restingstate.py

```diff
@@ -178,10 +197,22 @@ def compute_alff(data_matrix, low_pass, high_pass, TR, sample_mask=None):
         # from the value closest to the low pass cutoff, to the value closest
         # to the high pass pass cutoff
         alff[i_voxel] = len(ff_alff) * np.mean(power_spectrum_sqrt[ff_alff[0] : ff_alff[1]])
+        falff[i_voxel] = np.sum(power_spectrum_sqrt[ff_alff[0] : ff_alff[1]]) / np.sum(
+            power_spectrum_sqrt
+        )
+        peraf[i_voxel] = (
+            np.mean(np.abs((voxel_data_for_peraf - mean_matrix[i_voxel]) / mean_matrix[i_voxel]))
+            * 100
+        )
+
+        tsnr[i_voxel] = mean_matrix[i_voxel] / np.std(voxel_data_for_peraf)
```

Comment:
Does this ever produce nans or infs?

Reply:
I have a check above for a zero-variance time series that will set the voxel's value to 0 instead of running the equation, so it shouldn't produce any NaNs or Infs.

Reply:
sorry I missed it! In that case I approve